source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -284,6 +284,12 @@ class Filter : public Logger::Loggable<Logger::Id::ext_proc>,
 
   void sendTrailers(ProcessorState& state, const Http::HeaderMap& trailers);
 
+  Http::LocalErrorStatus onLocalReply(const LocalReplyData&) override {
```

Comment:
I'm pondering the experience of an ext_proc server in this situation. Let's say the router decides to 5xx and we get a local reply. We've sent some headers for the request to the ext_proc server and now we're about to send 5xx to the client. Surely we should send _some_ message to the ext_proc server to indicate what is going on? Maybe an explicit local reply message?

Reply:
Connection reset will destroy ext_proc filter, and close the gRPC stream. So, ext_proc server will get a notification from gRPC stream close. Is that something sufficient?  If this is acceptable, then the change is simple as proposed in the PR.

If this is not acceptable, and an explicit local reply message needs to be sent to ext_proc server onLocayReply(),  I guess we can do:

In the encodeHeaders(), if it is localReply message, then send the message to the ext_proc server,  and return FilterHeadersStatus::Continue. This is different from normal encodeHeaders() processing which send the headers, and return FilterHeadersStatus::StopIteration, which cause the header is held up in the filter chain.
This way, onLocalyReplay(), the headers is sent to both ext_proc server and downstream client.  And Envoy do not need to wait for the response from the ext_proc server before the connection get closed.

Reply:
I guess I prefer to err towards having more observability than less, that said, I'd be OK with this becoming a TODO (with tracking issue filed in GH) to avoid bloating the PR.

Reply:
The `onLocalReply` callback is provided a hint indicating the request is going to be immediately reset after encoder filter has completed. ext_proc must not pause filter chain if this flag is set, otherwise it will cause the reset without the response. ext_proc, can send a 'fire and forget' message to the ext_proc service with this response for observability, but it would need to deal with the complexity of not cancelling the message to the ext_proc in case it could not be immediately sent (i.e. there is no flow control window). I'm actually not sure if Envoy can deal with all these corner cases.

I think there are two possibilities:

1. Not send anything to ext_proc if `reset_imminent_ == true`
2. Fire and forget a message in this case, without pausing the filter chain and accept for now that some messages in this case may be dropped before they are sent. File an Issue and deal with it later.